src/grid/grid-header.component.ts

```diff
@@ -138,7 +138,12 @@ export class IgxGridHeaderComponent implements IGridBus, OnInit, DoCheck {
     }
 
     get restrictResizeMin(): number {
-        const minWidth = Number.isNaN(parseFloat(this.column.minWidth)) ? 48 : parseFloat(this.column.minWidth);
+        const actualMinWidth = parseFloat(this.column.minWidth);
+        const defaultMinWidth  = parseFloat(this.column.defaultMinWidth);
+
+        let minWidth = Number.isNaN(actualMinWidth) || actualMinWidth < defaultMinWidth ? defaultMinWidth : actualMinWidth;
+        minWidth = minWidth < parseFloat(this.column.width) ? minWidth : parseFloat(this.column.width);
```

Comment:
Not for this PR: I think we shouldn't change the minWidth. Normally the idea of minWidth is to restrict the width of the column. We are currently doing the opposite. But this could be discussed and changed in the future.